pay-api/src/pay_api/services/statement.py

```diff
@@ -440,27 +443,46 @@ def get_statement_report(statement_id: str, content_type: str, **kwargs):
         else:
             report_name = f"{report_name}-{from_date_string}-to-{to_date_string}.{extension}"
 
-        statement_purchases = Statement.find_all_payments_and_invoices_for_statement(statement_id)
-
-        result_items = InvoiceSearch.create_payment_report_details(purchases=statement_purchases, data=None)
         statement = statement_svc.asdict()
         statement["from_date"] = from_date_string
         statement["to_date"] = to_date_string
 
-        report_inputs = PaymentReportInput(
-            content_type=content_type,
-            report_name=report_name,
-            template_name=StatementTemplate.STATEMENT_REPORT.value,
-            results=result_items,
-        )
+        if extension == "csv":
+            statement_purchases = Statement.find_all_payments_and_invoices_for_statement(statement_id)
+            result_items = InvoiceSearch.create_payment_report_details(purchases=statement_purchases, data=None)
+            report_inputs = PaymentReportInput(
+                content_type=content_type,
+                report_name=report_name,
+                template_name=StatementTemplate.STATEMENT_REPORT.value,
+                results=result_items,
+            )
 
-        summary = Statement._build_statement_summary_for_methods(statement_dao, statement_purchases)
-        if summary:
-            report_inputs.statement_summary = summary
+            report_response = InvoiceSearch.generate_payment_report(
+                report_inputs, auth=kwargs.get("auth", None), statement=statement
+            )
+        else:
+            statement_to_date = statement_svc.to_date
+
+            invoices_orm = Statement.find_all_payments_and_invoices_for_statement(
+                statement_id, include_partial_refunds_credits=True
+            )
+
+            invoice_ids = [inv.id for inv in invoices_orm]
+            db_summaries = Statement.get_totals_by_payment_method_from_db(invoice_ids, statement_to_date)
```

Comment:
you're grabbing results, building an object then putting them into a query

that's not going to work (wont scale), like I said before if you return query and not query.all(), you can use subquery and do a select based on the `statement_invoices` query